src/s3ql/fs.py

```diff
@@ -57,6 +57,17 @@
 else:
     ACL_ERRNO = errno.EOPNOTSUPP
 
+class _UpdateSize:
+    """A private class used for O_TRUNC.
+
+    This is needed because pyfuse3.SetattrFields fields are read only"""
+    def __init__(self):
```

Comment:
To make this instances more lightweight, I'd add a `__slots__ = ['update_atime', ...]` definition.

Reply:
How about the the current implementation? I forgot to also set the `mtime_ns` – that should be set to the current time when a file is opened with O_TRUNC. Since this object only gets read, I just created one global instance of it.